tests/unit/test_vcr.py

```diff
@@ -335,7 +335,7 @@ def function_with_return():
     assert ret == function_with_return()
 
 
-class TestVCRClass(VCR().test_case()):
+class TestVCRClass(VCR()):
```

Comment:
@jairhenrique this change seems broken to me (all else looks great). Could you elaborate on the idea behind this change? I would expect some form of inlining of the original auto-wrapping with `VCR.use_cassette` but this new version now derives from an instance of `VCR` rather than a class. Please help me understand the idea. Thanks!

Reply:
@hartwork I just erased things without thinking too much. As the tests are passing I thought everything was fine. I'll go back to the implementation and look at it calmly.

Reply:
@jairhenrique thanks! Maybe we even need to keep `def test_case` around (and get the six bits out of it or not). This could be of interest: https://stackoverflow.com/a/18513858/11626624 . I haven't fully wrapped my head around it yet.

Reply:
@jairhenrique a few more thoughts if you don't mind:
- We could bring back `VCR.test_case` and six in for now and treat it as a separate issue.
- `VCR.test_case` could be argued to be part of public API. It's maybe the only bit in here that could be argued to break the API.
- Using six' `with_metaclass` may be nicer than inlining it (if keeping `VCR.test_case` intact is the goal).

See you

Reply:
@jairhenrique I was able to verify the regression in practice just mow:

#### Before on `master`

```console
# tox -e py310-requests-urllib3-1 -- -v tests/unit/test_vcr.py::TestVCRClass | grep ::test
tests/unit/test_vcr.py::TestVCRClass::test_one PASSED                    [ 33%]
tests/unit/test_vcr.py::TestVCRClass::test_two PASSED                    [ 66%]
tests/unit/test_vcr.py::TestVCRClass::test_dynamically_added PASSED      [100%]
```

#### After on `drop37`

```console 
# tox -e py310-requests-urllib3-1 -- -v tests/unit/test_vcr.py::TestVCRClass | grep ::test
<no output>
```

Probably the main question now is whether we want to bring the dependency on six back for [`six.with_metaclass`](https://github.com/benjaminp/six/blob/3b7efbcca41857da03fb01f004ccc425ab82dfbf/six.py#L856-L877) or include a copy of that function in VCR.py (e.g. named `_with_metaclass`). The license would allow it, but personally I'm in favor of using `six.with_metaclass` for this. What do you think?

Reply:
@hartwork I revert this changes.